### PR TITLE
Update GRANT re: v22.2+ DB privilege inheritance

### PR DESCRIPTION
**Pull request by rmloveland**
Fixes DOC-3192

Summary of changes:

- Update `GRANT` docs:

  - Note that when a role/user is granted privileges for a database, new tables will *not* inherit those privileges

  - Show how to grant all privileges on existing tables to User X

  - Show how to make sure when a new table is created, all privileges on that table are granted to User X

  - Update some SQL outputs to match the new reality

  - Sprinkle some `IF NOT EXISTS` on a few SQL statements to make them ~idempotent for easier testing

### CODE REVIEW

**Review by rafiss — APPROVED** [review]
this lgtm factually! just had some optional suggestions

File: v22.2/grant.md [hunk 1]
```diff
@@ -49,15 +49,11 @@ Roles and users can be granted the following privileges:
 
 ### Granting privileges
 
-- When a role or user is granted privileges for a database, new tables created in the database will inherit the privileges, but the privileges can then be changed.
+When a role or user is granted privileges for a table, the privileges are limited to the table. The user does _not_ automatically get privileges to new or existing tables in the database. To grant privileges to a user on all new and/or existing tables in a database, see [Grant privileges on all tables in a database](#grant-privileges-on-all-tables-in-a-database-or-schema).
 
-    {{site.data.alerts.callout_info}}
-    The user does not get privileges to existing tables in the database. To grant privileges to a user on all existing tables in a database, see [Grant privileges on all tables in a database](#grant-privileges-on-all-tables-in-a-database-or-schema)
-    {{site.data.alerts.end}}
+The `root` user automatically belongs to the `admin` role and has the `ALL` privilege for new databases.
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply v22.2/grant.md]: I moved this info up to the beginning of the page (around line 10) and revised it a bit to link to some stuff and hopefully it says about what your comment is saying

PTAL and let me know if not though!  Still learning in the roles/privs area

File: v22.2/grant.md [hunk 2]
```diff
@@ -141,29 +132,62 @@ SHOW GRANTS ON TABLE movr.public.*;
...
   movr          | public      | vehicles                   | demo    | ALL            |      t
-  movr          | public      | vehicles                   | max     | SELECT         |      f
+  movr          | public      | vehicles                   | max     | ALL            |      f
+  movr          | public      | vehicles                   | public  | SELECT         |      f
   movr          | public      | vehicles                   | root    | ALL            |      t
-(24 rows)
+(26 rows)
+~~~
+
+To ensure that anytime a new table is created, all the privileges on that table are granted to a user:
```
[the remark this answers is not in the document]
Comment (rmloveland) [reply v22.2/grant.md]: added a link to `ALTER DEFAULT PRIVILEGES` docs

also went and added some links to the SQL statements used in the next couple of steps as well

**Comment by rmloveland** [comment]
TFTR, Rafi! 

Gonna pass along to Ian for Docs team review, please shout if my latest changes introduced any subtle factual errors (although hopefully not!)

**Review by rafiss — APPROVED** [review]
this lgtm! thanks

**Review by ianjevans — APPROVED** [review]

**Pull request merged**